sbin/upload-artifacts

```diff
@@ -1,151 +1,189 @@
+                ARTIFACTS+=("$1")
+                ;;
+        esac
+        shift
+    done
+}
 
-#----------------------------------------------------------------------------------------------
+# Initialize variables
+ARTIFACTS=()
 
-ARCH=$(uname -m)
-
-[[ $ARCH == arm64 ]] && ARCH="aarch64"
-[[ $ARCH == x64 ]] && ARCH="x86_64"
-
-OS=$($READIES/bin/platform --os)
-[[ $OS == linux ]] && OS="Linux"
-
-[[ -z $OSNICK ]] && OSNICK=$($READIES/bin/platform --osnick)
-[[ $OSNICK == trusty ]]  && OSNICK=ubuntu14.04
-[[ $OSNICK == xenial ]]  && OSNICK=ubuntu16.04
-[[ $OSNICK == bionic ]]  && OSNICK=ubuntu18.04
-[[ $OSNICK == focal ]]   && OSNICK=ubuntu20.04
-[[ $OSNICK == jammy ]]   && OSNICK=ubuntu22.04
-[[ $OSNICK == noble ]]   && OSNICK=ubuntu24.04
-[[ $OSNICK == centos7 ]] && OSNICK=rhel7
-[[ $OSNICK == centos8 ]] && OSNICK=rhel8
-[[ $OSNICK == centos9 ]] && OSNICK=rhel9
-[[ $OSNICK == ol8 ]]     && OSNICK=rhel8
-[[ $OSNICK == rocky8 ]]  && OSNICK=rhel8
-[[ $OSNICK == rocky9 ]]  && OSNICK=rhel9
-[[ $OSNICK == alpine3* ]]  && OSNICK=alpine3
-
-
-# Fix OSNICK for NotpineForGHA3 (alpine workaround to support arm64 GH Actions)
-if [[ $OSNICK == NotpineForGHA3* ]]; then
-	OSNICK=alpine3
+# Check for help flag in environment variable
+if [[ $HELP == 1 ]]; then
+    print_help
+    exit 0
 fi
 
-PLATFORM="$OS-$OSNICK-$ARCH"
+# Parse command line arguments
+parse_args "$@"
 
-#----------------------------------------------------------------------------------------------
+# If no specific artifacts provided, we'll use defaults later
+
+ARCH=$($GET_PLATFORM --arch)
+[[ $ARCH == x64 ]] && ARCH=x86_64
+[[ $ARCH == arm64v8 ]] && ARCH=aarch64
+
+OS=$($GET_PLATFORM --os)
+[[ $OS == linux ]] && OS=Linux
+
+OSNICK=$($GET_PLATFORM --version-artifact)
+
+PLATFORM="$OS-$OSNICK-$ARCH"
+echo "Detected OS: $OS"
+echo "Detected OSNICK: $OSNICK"
+echo "Detected ARCH: $ARCH"
+echo "Detected PLATFORM: $PLATFORM"
 
 OP=""
 [[ $NOP == 1 ]] && OP=echo
+S3_URL=s3://redismodules
 
-if [[ $STAGING == 1 ]]; then
-	S3_URL=s3://redismodules/lab/staging
-else
-	S3_URL=s3://redismodules
+
+if [[ -z $GITHUB_ACTIONS && $FORCE != 1 ]]; then
+    eprint "Cannot upload outside of GitHub Actions. Override with FORCE=1."
+    exit 1
 fi
 
-if [[ $FORCE != 1 ]]; then
-	# TODO: remove circle
-	if [[ -z $GITHUB_ACTIONS && -z $CIRCLECI ]]; then
-		eprint "Cannot upload outside of GitHub Actions. Override with FORCE=1."
-		exit 1
-	fi
-
-	if [[ -z $AWS_ACCESS_KEY_ID || -z $AWS_SECRET_ACCESS_KEY ]]; then
-		eprint "No credentials for S3 upload."
-		exit 1
-	fi
+if [[ -z $AWS_ACCESS_KEY_ID || -z $AWS_SECRET_ACCESS_KEY ]]; then
+	eprint "No credentials for S3 upload."
+	exit 1
 fi
 
-cd $ROOT/bin
 
-if [[ $SNAPSHOT == 1 ]]; then
-	MAYBE_SNAP=/snapshots
-else
-	MAYBE_SNAP=
-fi
+#----------------------------------------------------------------------------------------------
+# Navigate to artifacts directory
+cd "$ROOT/bin"
+MAYBE_SNAP="/snapshots"
```

Comment:
why do we need this `MAYBE_SNAP`? aren't we always uploading to `snapshots` dir?